Created by Jasper Grimm (UoY)
Modified for 1.3.8 by Pierre Berthier (Unibe)
Install header files and scripts
Rename static lib to freebayes, and set install to True 
Add target to build shared library
diff -Nru freebayes-1.3.8/meson.build freebayes-1.3.8b/meson.build
--- freebayes-1.3.8/meson.build	2024-06-13 11:36:05.940969748 +0200
+++ freebayes-1.3./meson.build	2024-06-13 11:42:01.001963517 +0200
@@ -203,6 +203,38 @@
 freebayes_src = files('src/freebayes.cpp')
 bamleftalign_src = files('src/bamleftalign.cpp')

+# FreeBayes headers
+freebayes_headers = files(
+    'src/Allele.h',
+    'src/AlleleParser.h',
+    'src/BedReader.h',
+    'src/Bias.h',
+    'src/CNV.h',
+    'src/Contamination.h',
+    'src/DataLikelihood.h',
+    'src/Dirichlet.h',
+    'src/Ewens.h',
+    'src/FBFasta.h',
+    'src/Genotype.h',
+    'src/IndelAllele.h',
+    'src/LeftAlign.h',
+    'src/LargeFileSupport.h',
+    'src/Logging.h',
+    'src/Marginals.h',
+    'src/Multinomial.h',
+    'src/NonCall.h',
+    'src/Parameters.h',
+    'src/Product.h',
+    'src/Result.h',
+    'src/ResultData.h',
+    'src/Sample.h',
+    'src/SegfaultHandler.h',
+    'src/Sum.h',
+    'src/TryCatch.h',
+    'src/Utility.h',
+    'src/version_git.h',
+    )
+
 # Include paths
 incdir = include_directories(
     'src',
@@ -267,14 +267,27 @@
     )
 
 freebayes_lib = static_library(
-    'freebayes_common',
+    'freebayes',
     freebayes_common_src,
-    include_directories : [multichoose_inc, fastahack_inc, smithwaterman_inc, incdir],
+    include_directories : [multichoose_inc, fastahack_inc, smithwaterman_inc, intervaltree_inc, incdir],
     cpp_args : extra_cpp_args,
     dependencies : [zlib_dep, lzma_dep, thread_dep, htslib_dep, tabixpp_dep,
                     vcflib_dep, wfa2lib_dep, seqlib_dep, fastahack_dep, smithwaterman_dep],
-    install : false,
+    install : true,
+    )
+freebayes_slib = shared_library(
+    'freebayes',
+    freebayes_common_src,
+    include_directories : incdir,
+    cpp_args : extra_cpp_args,
+    dependencies : [zlib_dep, lzma_dep, thread_dep, htslib_dep, tabixpp_dep,
+                    vcflib_dep, seqlib_dep, fastahack_dep, smithwaterman_dep],
+    install : true,
     )
 
+install_headers(freebayes_headers, subdir : 'freebayes')
+
+install_subdir('scripts', install_dir : '')
+
 if static_build
   link_arguments = '-static'
 else
@@ -292,7 +294,7 @@
 
 executable('freebayes',
            [freebayes_src,multichoose_src,fastahack_src,smithwaterman_src],
-           include_directories : [multichoose_inc, fastahack_inc, smithwaterman_inc, incdir],
+           include_directories : [multichoose_inc, fastahack_inc, smithwaterman_inc, intervaltree_inc, incdir],
            cpp_args : extra_cpp_args,
            link_args: link_arguments,
            dependencies: [zlib_dep,
@@ -311,7 +313,7 @@
 
 executable('bamleftalign',
            [bamleftalign_src,fastahack_src,smithwaterman_src],
-           include_directories : [multichoose_inc, fastahack_inc, smithwaterman_inc, incdir],
+           include_directories : [multichoose_inc, fastahack_inc, smithwaterman_inc, intervaltree_inc, incdir],
            cpp_args : extra_cpp_args,
            link_args: link_arguments,
            dependencies: [zlib_dep,

